.github/workflows/test-framework.yml

```diff
@@ -18,25 +18,41 @@ on:
 
 jobs:
   test:
-    name: run tests
+    name: ${{ matrix.os }}, node ${{ matrix.node-version }}, ${{ matrix.browser }}
     runs-on: ${{ matrix.os }}
+    if: "!contains(github.event.head_commit.message, 'skip ci')"
 
     strategy:
       matrix:
-        os: [ubuntu-latest, windows-latest]
+        os: [ubuntu-latest, windows-latest, macos-latest]
         node-version: [10.x, 12.x, 14.x]
+        browser: [no-browser, chromium, firefox, webkit]
```

Comment:
if you change this to on the whole handling will be done in compile.js.
The later testing is not neccesary then.